packages/compass-editor/src/types.d.ts

```diff
@@ -0,0 +1,58 @@
+import type AceBuilds from 'ace-builds';
+import type { Ace } from 'ace-builds';
+
+type GlobalAce = {
+  EditSession: typeof AceBuilds['EditSession'];
+  Range: typeof AceBuilds['Range'];
+  UndoManager: typeof AceBuilds['UndoManager'];
+  VirtualRenderer: typeof AceBuilds['VirtualRenderer'];
+  config: typeof AceBuilds['config'];
+  createEditSession: typeof AceBuilds['createEditSession'];
+  edit: typeof AceBuilds['edit'];
+  require: typeof AceBuilds['require'];
+  version: typeof AceBuilds['version'];
+  acequire: typeof AceBuilds['require'];
+  define: (
+    module: string,
+    depdendencies: string[],
+    moduleFn: (
+      acequire: typeof AceBuilds['require'],
+      exports: any,
+      module: any
+    ) => void
+  ) => void;
+};
+
+declare global {
+  declare const ace: GlobalAce;
+}
+
+export interface HighlightRules {
+  normalizeRules(): void;
+  createKeywordMapper(
+    map: Record<string, string>,
+    defaultToken: string
+  ): (...args: any[]) => any;
+  $rules: Record<
+    string,
+    {
+      regex: string | RegExp;
+      token: string | string[] | ((...args: any[]) => any);
+      next?: string;
+    }[]
+  >;
+}
+
+export interface WorkerClient {
+  attachToDocument(doc: Document): void;
+}
+
+export interface AceMode {
+  HighlightRules: new (...args: unknown) => HighlightRules;
+  createWorker(session: Ace.EditSession): WorkerClient | null;
+}
+
+export type MongoDBCompletion = Ace.Completion & {
```

Comment:
A bit of a noob question, is there a reason why we have to define this type here vs a regular `.ts` file? Makes it seems special to me 🤷‍♂️

Reply:
Ah, I'll change it! I started initially with making them globally available (because ace needs to be defined in global scope anyway) for which you'd usually use a d.ts, but had to convert it at some point, because you can't mix imports and global type definitions

Reply:
Done in https://github.com/mongodb-js/compass/pull/3589/commits/562c7b1287336e64d71281a249dc970e097eee28